### ISSUE TITLE
locale id of moment is different from cldr

### ISSUE THREAD

**Issue by Maroon1** [angular/material2#8268]
#### Bug, feature request, or proposal:
Bug

#### What is the current behavior?
MomentDateAdapter use LOCAL_ID by default
```javascript
  constructor(@Optional() @Inject(MAT_DATE_LOCALE) dateLocale: string) {
    super();
    this.setLocale(dateLocale || moment.locale());
  }
```
but,[ angular](https://github.com/angular/angular/tree/master/packages/common/locales) use [cldr ](http://cldr.unicode.org/core-spec#Unicode_Language_and_Locale_Identifiers) spec from v5. they are different.
for example: 

|  locale |  CLDR | Moment |
| - | - | -|
| chinese | zh-Hans | zh-cn |

cause moment.localeData can't find the locale data with that locale id.

#### What are the steps to reproduce?
Providing a StackBlitz/Plunker (or similar) is the *best* way to get the team to see your issue. <br/>
Plunker starter (using on `@master`): https://goo.gl/uDmqyY<br/>
StackBlitz starter (using latest `npm` release): https://goo.gl/wwnhMV<br/>

**Comment by mmalerba** [comment]
@vicb is there any way for us to get the old style locale ID? It seems Moment.js is not compatible with the BCP47 ones.

**Comment by vicb** [comment]
@mmalerba Angular should use whatever you ask him to use, no ? 
I guess I'm missing context here. How is the app built ?

**Comment by vicb** [comment]
FYI this issue might be linked to angular/angular#20340 ?
Seems like a webpack plugin is trying to be too smart. @ocombe is working on it and will update this issue.
In the meantime @Maroon1 could you please confirm what your build process is. Thanks.

**Comment by Maroon1** [comment]
[demo](https://github.com/Maroon1/material2-moment-adapter-issue), run `npm run serve:aot` @vicb.

**Comment by vicb** [comment]
So @ocombe I guess what's relevant here is 

> "serve:aot": "ng serve --aot --local zh-hans --open",

ie CLI app. Thanks for the details @Maroon1

**Comment by ocombe** [comment]
Hello, we've reviewed the issue and we think that we can do to things to fix this:
- update the i18n docs with the list of equivalent locales (like zh-Hans / zh-cn), you can already find this list in our extraction scripts if you're curious: https://github.com/angular/angular/blob/master/tools/gulp-tasks/cldr/closure.js#L35
- let user register locale data along with the locale id of their choice (if you still want to use the old locale id but need to import the locale data from angular)

**Comment by mmalerba** [comment]
@vicb @ocombe so am I correct in saying that there's nothing for material to do here other than wait for the fix in core?

**Comment by vicb** [comment]
The fix has been merged in the core and will be available in the 5.1 release branch.

**Comment by mmalerba** [comment]
excellent, closing this issue then
